lib/packwerk/folder_visibility/checker.rb

```diff
@@ -0,0 +1,93 @@
+# typed: strict
+# frozen_string_literal: true
+
+require 'packwerk/folder_visibility/package'
+require 'packwerk/folder_visibility/validator'
+
+module Packwerk
+  module FolderVisibility
+    class Checker
+      extend T::Sig
+      include Packwerk::Checker
+
+      VIOLATION_TYPE = T.let('folder_visibility', String)
+
+      sig { override.returns(String) }
+      def violation_type
+        VIOLATION_TYPE
+      end
+
+      sig do
+        override
+          .params(reference: Packwerk::Reference)
+          .returns(T::Boolean)
+      end
+      def invalid_reference?(reference)
+        constant_package = reference.constant.package
```

Comment:
Does `reference.package` contain the reference, and is `reference.constant` the pack that `reference.package` refers to?

Reply:
I pushed an update to clarify this code